### DISCUSSION TITLE
High CPU usage for conversion to AVIF

### DISCUSSION THREAD

**Discussion by StonyBrookReus** [ImageMagick/ImageMagick#5128] [Help]
I have observed very high CPU usage(200%-300%) while running commands to covert images to AVIF consecutively, no matter what the input was. It didn't happen for JPG, so I think it's for AVIF only. Is there any method to fix it?

IM version: `7.1.0-20`
Ubuntu version: `Ubuntu 18.04.5 LTS`
Command: `convert fox.profile0.8bpc.yuv420.avif -quality 80 -filter triangle -resize 1600x1600^> -unsharp -strip test2.avif`

Please let me know if you need more details. Thanks!

**Comment by urban-warrior** [comment]
ImageMagick leverages the `ffmpeg` delegate to write in the AVIF format.  We assume the high CPU utilization may come from `ffmpeg` rather than ImageMagick.

**Comment by snibgo** [comment]
@StonyBrookReus: I don't suppose your command works, as "-unsharp" needs an argument. See https://imagemagick.org/script/command-line-options.php#unsharp

You should re-run your test without the resize and unsharp. Those operations are performed by IM code, and may be CPU intensive.

For operations internal to IM, you can reduce CPU intensity with, for example, "-limit throttle 1000 -limit thread 1".
